src/NUnitTestAdapter/TestConverter.cs

```diff
@@ -221,6 +221,30 @@ private VSTestResult GetBasicResult(XmlNode resultNode)
 
             vsResult.ComputerName = Environment.MachineName;
 
+            foreach (var output in outputNodes)
+            {
+                var stream = output.GetAttribute("stream");
+                if (string.IsNullOrEmpty(stream))
+                {
+                    continue;
+                }
+
+                var text = output.InnerText;
+                string category;
+                switch (stream.ToLower())
+                {
+                    case "error":
+                        category = TestResultMessage.StandardErrorCategory;
+                        break;
+
+                    default:
+                        category = TestResultMessage.StandardOutCategory;
+                        break;
+                }
+
+                vsResult.Messages.Add(new TestResultMessage(category, text));
+            }
+
             XmlNode outputNode = resultNode.SelectSingleNode("output");
             if (outputNode != null)
```

Comment:
I am not sure I understand

Reply:
Let's assume your test ONLY writes to the error stream.  Then the resultnode , which comes from the test results, will not contain any output node.  It will be null.  But you still have this single Error text you want to add to the output node.  So you must create a new XmlNode to contain this error text and add that in this case.

Reply:
@OsirisTerje sorry, I didn't catch the thought. If the result node does not contain the output node the method vsResult.Messages.Add() is not invoked for this output, but anyway this method returns vsResult containing all error/progress messages inside.

Reply:
Ok, seems you're right.  Walked through the code there. :-)